lib/net/peer.js

```diff
@@ -1751,32 +1750,6 @@ class Peer extends EventEmitter {
     this.send(new packets.MempoolPacket());
   }
 
-  /**
-   * Send `reject` to peer.
-   * @param {Number} code
-   * @param {String} reason
-   * @param {String} msg
-   * @param {Hash} hash
-   */
-
-  sendReject(code, reason, msg, hash) {
-    const reject = packets.RejectPacket.fromReason(code, reason, msg, hash);
-
-    if (msg) {
-      this.logger.debug('Rejecting %s %h (%s): code=%s reason=%s.',
```

Comment:
I think it's important that we continue to log reject reasons, even if we don't send the packets.

Reply:
We may already be logging it internally, I'll check though.

Reply:
Perhaps `increaseBan` should log the reason, each time.

Reply:
> I think it's important that we continue to log reject reasons, even if we don't send the packets.

I agree with this.

> Perhaps increaseBan should log the reason, each time.

This seems like a smart place to log. I would like to know when my peers misbehave.

Reply:
Except now that I think about it, that could be a DOS vector regarding a disk fill attack when too many loglines

Reply:
Useful to consider, however it likely wouldn't be significant as at maximum it would be 100 lines, with increasing the score at 1 per time.

Reply:
To allow for logging the reason, we would need to change the function signature of `peer.increaseBan`. Right now it just takes a ban score to add, the second argument could be a `msg`.

```javascript
this.logger.info('Increasing banscore for %s by %d: %s', this.hostname(), score, msg);
```

What do you think?

Reply:
I am wondering about the DoS point, as there could be something there not considered.

Reply:
Maybe the logger should log on the `debug` level when logging about banscores. It would prevent possible attacks if we assume that production nodes should run with their logger on `info`

Reply:
Default logger level is `debug`:

https://github.com/bcoin-org/bcoin/blob/f991b0a60c575610cdbcdfe2205316a2012a61f5/bin/node#L30